pinot-spi/src/main/java/org/apache/pinot/spi/auth/TableAuthorizationResult.java

```diff
@@ -0,0 +1,79 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.spi.auth;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import org.apache.commons.lang.StringUtils;
+
+
+/**
+ * Implementation of the AuthorizationResult interface that provides authorization results
+ * at the table level, including which tables failed authorization.
+ */
+public class TableAuthorizationResult implements AuthorizationResult {
+
+  private static final TableAuthorizationResult SUCCESS = new TableAuthorizationResult(Set.of());
+  private final Set<String> _failedTables;
+
+  public TableAuthorizationResult() {
+    _failedTables = new HashSet<>();
+  }
+
+  public TableAuthorizationResult(Set<String> failedTables) {
+    _failedTables = new HashSet<>(failedTables);
```

Comment:
I don't think we need to clone a set
```suggestion
    _failedTables = failedTables;
```

Reply:
I made a clone as earlier we had `addFailedTable()` which can edit _failedTables and there by modifying the API caller's failedTable set , but as we removed `addFailedTable()` , it makes sense to remove now .